### ISSUE TITLE
UX Concepts for soundtrack.io 

### ISSUE THREAD

**Issue by alexknutson** [FabricLabs/soundtrack#21]
I've been working on some UX ideas guys. Would love some feedback!

**Disclaimer:**  _This is by no means a completed design. No design is ever really "finished", but it's still missing key components to the UI..._
### Main Concept -

![concept-2-feedback](https://f.cloud.github.com/assets/905228/819088/86eba37a-efaa-11e2-9d22-2802880b114a.png)
### Concept v1 - Where Add Track overlaps the main content

![concept-2-feedback-add-track-sidebar](https://f.cloud.github.com/assets/905228/819075/0cb33654-efaa-11e2-89e6-f34931e5dc52.png)
### Concept v2 - Where Add Track is "inserted" between the Playlist Queue and the Video Player. This briefly pushes content off the screen for some small resolutions.

![concept-2-feedback-add-track-sidebar-insert](https://f.cloud.github.com/assets/905228/819093/a9733570-efaa-11e2-9c2c-b90a0060d60a.png)
# List of things to do:
- [ ] I'm not attached to the 'spin it, lame' language. My goal was to visualize the amount of votes the track has received (the green/red bars increase in width as the votes accumulate)
- [x] I think we can do something more interesting with our logo. Though I have no idea where to go with that yet. For now, it's similar to what we currently have.
- [ ] Volume/Mute Icon next to the volume slider
- [ ] Create the Add Track button that triggers the Add Track search/import menu.
- [ ] More sample tracks in the queue to visualize how a fully populated queue might look like.
- [ ] The 'Logged In' user settings bar on the far left side needs to be figured out. SMALL menu that is heavily icon based. Maybe no text at all? ie: Gear for account settings, logout button, another manage playlist icon (similar to the shortcut found above the playing video), etc
- [ ] More examples of what each page (About, History, People) might look like using the overall design. I really like the minimalism of "inserting" those page's content the way the "Add Track" content is inserted into the main view. Ideally--- I don't think we ever want to take the user away from the main page. If we can keep the content of those pages short and sweet, maybe we can fit it all in a similar way?
# Take note of...

![concept-2-feedback - copy](https://f.cloud.github.com/assets/905228/819025/345309ac-efa8-11e2-9566-89e4378905b5.png)
- [ ] The heart (shown as active) should toggle the Add to Playlist fancybox (or whatever it is we use), and slide up the text above to reflect when the user first added this song to their specified playlist.
- [ ] Google+ Sign in? Please? :+1: 
- [ ] Eric mentioned he likes Blynde's sidebar for the songs queue. I made the design with the left sidebar in mind, but this could easily be reworked so that the sidebars are fixed to the right. 
## The Geolocation Visualizer

Obviously this would take significant work to bring into fruition, but I think it would be slick to visualize where users are coming from, their profile, and what our peak hours are for each day. If you guys don't like that idea, ignore the entire bottom half of the main content area. 
![concept-2-feedback-add-track-sidebar-insert - copy](https://f.cloud.github.com/assets/905228/818989/d1780ee6-efa6-11e2-8447-c0af498539b8.png)

**Comment by martindale** [comment]
I've opened a new milestone, [UI Overhaul](https://github.com/martindale/soundtrack.io/milestones/UI%20Overhaul), to track all the issues related to this.

The current interface is built, poorly, on top of Bootstrap 2 (see #115).  We should evaluate a new framework, with the following goals in mind;

- fully responsive (#76)
- clean, modern style (#21)
- single page application (#31)

I'd like to investigate the following technologies:
- [Semantic UI](http://semantic-ui.com/) instead of Bootstrap
- [Flexbox](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Flexible_boxes)
- [Durandal](https://github.com/BlueSpire/Durandal)
- [Maki](https://github.com/martindale/maki)

I know @alexknutson and @vorathe have expressed interest in working on this, and I'd like some help as we investigate what it will take to turn soundtrack from its current prototype state into a usable public beta for others.

**Comment by ZachCase** [comment]
I will definitely be helping with this project. I am currently working on the landing page UI overhaul. I am mocking it up in Sketch, currently. I will post the mock-up when I am finished and then we can go from there!

**Comment by ZachCase** [comment]
@martindale I think that Flexbox would be a great technically to use with the site's Front-end. PureCSS is a grid-framework that uses flexbox, so that would be something to look into.

**Comment by martindale** [comment]
@ZachCase I agree totally that Flexbox is the best approach for the layout.  I've half considered just starting a clean Flexbox layout from scratch, and then writing my own quick little two-way databinding to manage it.  Haven't had time.  Keep up the good work!

**Comment by ZachCase** [comment]
@Vorathe Here is the mock-up I made. The only thing I left out was the footer... due to laziness, sorry.

![desktop](https://cloud.githubusercontent.com/assets/7879141/7580907/f23b1e0a-f839-11e4-822b-1be704aa0cf1.png)

**Comment by allusis** [comment]
From scratch is probably the best option, but there are a few nice alternatives worth considering:

The Bootstrap4 alpha has flexbox opt-in, which is pretty good if you are semi-used to bootstrap markup. Of corse using the Sass version, and removing all of the unused components helps quite a bit with the bloat.

Also, if no one has looked into the Angular Material lately, it's come a long way. 
https://material.angularjs.org/latest/#/

**Comment by martindale** [comment]
Awesome feedback, @allusis.  Have you seen [Semantic UI](http://semantic-ui.com/)?  I'm using it in [Maki](https://maki.ericmartindale.com), and I think it might be a good foundation.  On the other hand, maybe a handwritten CSS stack might be even better.
